src/main/java/core/basesyntax/service/operation/impl/PurchareOperationHandler.java

```diff
@@ -0,0 +1,12 @@
+package core.basesyntax.service.operation.impl;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.operation.OperationHandler;
+
+public class PurchareOperationHandler implements OperationHandler {
+    @Override
+    public int calculationData(int balance, FruitTransaction transaction) {
+        return balance - transaction.getQuantity();
```

Comment:
What if balance of fruit is negative?
You have to do appropriate checks in your handlers.
For example,
```
int newQuantity = Storage.storage.get(fruit) - numbers;
if (newQuantity < 0) {
    throw new RuntimeException("Balance of fruit can't be negative");
}
Storage.storage.put(fruit, newQuantity);
```